### PR TITLE
Modify PowChain service

### PR DESCRIPTION
**Pull request by nisdas**
This PR is part of #1284. 

The goals of the PR are: 
- [x] Modify the PowChain Service to be up to date with the spec.
- [x] Update VRC contract.
- [x] Add Relevant tests

### CODE REVIEW

File: beacon-chain/powchain/service.go [hunk 1]
```diff
@@ -158,20 +195,61 @@ func (w *Web3Service) run(done <-chan struct{}) {
...
+
+	log.WithFields(logrus.Fields{
+		"publicKey": depositInput.Pubkey,
+	}).Info("Validator registered in VRC with public key")
+
+}
+
+func (w *Web3Service) SaveInTrie(depositData common.Hash, merkleRoot common.Hash) error {
+	if w.depositTrie.Root() != merkleRoot {
+		return errors.New("Saved root in trie is unequal to root received from log ")
```
Comment (prestonvanloon) [inline beacon-chain/powchain/service.go]: Please format this error properly. No capitalization, no trailing whitespace

**Comment by prestonvanloon** [comment]
Some lint issues

**Comment by nisdas** [comment]
@prestonvanloon Lint is fixed

File: beacon-chain/powchain/service.go [hunk 2]
```diff
@@ -158,20 +206,108 @@ func (w *Web3Service) run(done <-chan struct{}) {
...
+	index := binary.BigEndian.Uint64(MerkleTreeIndex)
+
+	log.WithFields(logrus.Fields{
+		"publicKey":         depositInput.Pubkey,
+		"merkle tree index": index,
+	}).Info("Validator registered in VRC with public key and index")
+
+}
+
+func (w *Web3Service) processChainStartLog(VRClog gethTypes.Log) {
```
Comment (prestonvanloon) [inline beacon-chain/powchain/service.go]:
```go
func (w *Web3Service) processChainStartLog(_ gethTypes.Log) {
  // TODO
}
```

File: beacon-chain/powchain/service.go [hunk 3]
```diff
@@ -158,20 +206,108 @@ func (w *Web3Service) run(done <-chan struct{}) {
...
+		w.processChainStartLog(VRClog)
+		return
+	}
+
+	log.Debugf("Log is not of a valid event signature %#x", VRClog.Topics[0])
+}
+
+// processDepositLog processes the log which had been received from
+// the POW chain by trying to ascertain which participant deposited
+// in the contract
```
Comment (terenc3t) [inline beacon-chain/powchain/service.go]: period.

File: beacon-chain/powchain/service.go (hunk 2 above)
Comment (terenc3t) [inline beacon-chain/powchain/service.go]: would be nice to add a comment here

File: beacon-chain/powchain/service.go [hunk 4]
```diff
@@ -158,20 +206,108 @@ func (w *Web3Service) run(done <-chan struct{}) {
...
+		"merkle tree index": index,
+	}).Info("Validator registered in VRC with public key and index")
+
+}
+
+func (w *Web3Service) processChainStartLog(VRClog gethTypes.Log) {
+	_ = VRClog
+}
+
+func (w *Web3Service) unPackLogData(data []byte) ([]byte, []byte, error) {
```
Comment (terenc3t) [inline beacon-chain/powchain/service.go]: comment here too

**Comment by nisdas** [comment]
@terenc3t @prestonvanloon Addressed your comments , PTAL 

@terenc3t We already have a prefix for the log https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/powchain/service.go#L16

File: beacon-chain/powchain/service.go (hunk 2 above)
Comment (prestonvanloon) [reply beacon-chain/powchain/service.go]: You will need to ignore the parameter name or the linter will complain it is unused. Please refer to the example I gave above.

Comment (prestonvanloon) [reply beacon-chain/powchain/service.go]: Nevermind, it looks like the linter doesn't care either way

**Pull request merged**